extensions/2.0/Khronos/KHR_materials_parallax_mapping/README.md

```diff
@@ -0,0 +1,59 @@
+<!--
+Copyright 2015-2022 The Khronos Group Inc.
+SPDX-License-Identifier: CC-BY-4.0
+-->
+
+# KHR\_materials\_parallax\_mapping
+
+## Contributors
+
+
+## Status
+
+Draft
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+Many materials provide a height-map that can either be used for real geometry displacement or for parallax-mapping.
+This extension defines a way to specify a texture as a height map.
+
+## glTF Schema Updates
+
+The height at a given position is calculated by sampling the given texture and using the red channel. The sampled value is then multiplied with the `heightFactor`.
+The result is used in an implementation defined way to offset all other texture lookups. (See Resources)
```

Comment:
It might be worthwhile to specify the direction of the offset.  (is white the topmost height or bottommost height?) Afaik Godot has white goes down, and calls the texture a "depth map"

Reply:
Good point! I will add that. Would it make sense to fix a certain behavior (i.e. 0 is lowest, 1 is highest) or have it configurable. Translating one to the other is trivial.

Alternatively, one could allow the `heightFactor` to be negative.

Not sure what is best. Any opinions/ideas?

Reply:
I spent the two last days on the question lol. Given it's a **height** map, of course white should be top (when height is at high value, it should be tall etc.). This goes against Godot's implementation (my own, and I imagine most people's), but it's just a question of doing `1.0 - height` in the shader, which I doubt would introduce an unacceptable overhead, especially given that parallax mapping algorithm oftentimes sample multiple times at different positions the height map per fragment.